### ISSUE TITLE
type(x) inaccurate for Unions

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#3108]
**Describe the bug**
Good behavior
```python
from __future__ import annotations


def f(x: int) -> None:
    reveal_type(type(x)) # type[int]
```

Incorrect behavior
```python
from __future__ import annotations


def f(x: int | str) -> None:
    reveal_type(type(x)) # Type revealed Any
```

**Expected behavior**
type[int | str] as revealed type. I checked with mypy and it shows Union[Type[builtins.str], Type[builtins.int]].

**VS Code extension or command-line**
I discovered this in pylance from hovering and then reproduced with cli on 1.1.224. I tried testing a couple older versions like 1.1.200/150/100 and they showed same behavior.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.225, which I just published. It will also be included in the next release of pylance.
